lib/search_files.js

```diff
@@ -0,0 +1,291 @@
+  return (typeof v === 'string');
+}
+
+function doNothing(v) {
+  return v;
+}
+
+
+function firstUnNullElem(arr, fn /*, ...*/) {
+  var args = Array.prototype.slice.call(arguments, 2);
+
+  for (var i = 0, n = arr.length; i < n; i++) {
+    var ret = fn.apply(arr, [arr[i]].concat(args));
+    if (ret) {
+      return ret;
+    }
+  }
+  return null;
+}
+
+function eachUnNullElem(arr, fn /*, ...*/) {
+  var args = Array.prototype.slice.call(arguments, 2);
+
+  var ret = [];
+  for (var i = 0, n = arr.length; i < n; i++) {
+    var v = fn.apply(arr, [arr[i]].concat(args));
+    if (v) {
+      if (isArray(v)) {
+        Array.prototype.push.apply(ret, v);
+      } else {
+        ret.push(v);
+      }
+    }
+  }
+  return ret;
+}
+
+function eachProp(obj, fn /*, ...*/) {
+  var args = Array.prototype.slice.call(arguments, 2);
+
+  var ret = {};
+  var keys = Object.keys(obj);
+  for (var i = 0, n = keys.length; i < n; i++) {
+    var key = keys[i];
+    ret[key] = fn.apply(obj, [obj[key], key].concat(args));
+  }
+  return ret;
+}
+
+
+function parsePath(pth) {
+  return path.parse ? path.parse(pth) : require('path-parse')(pth);
+}
+
+function isAbsolutePath(pth) {
+  return path.isAbsolute ? path.isAbsolute(pth) : !!parsePath(pth).root;
+}
+
+function normalizeHomedir(pth, homedir) {
+  if (!pth) {
+    return pth;
+  }
+
+  if (pth[0] === '~') {
+    if (pth.length === 1) {
+      return homedir;
+    }
+
+    if (pth[1] === '/') {
+      return path.join(homedir, pth.slice(2));
+    }
+
+    if (isWindows() && pth[1] === '\\') {
+      return path.join(homedir, pth.slice(2));
+    }
+  }
+
+  return pth;
+}
+
+function normalizePath(pth, ctx) {
+  if (isFunction(pth)) {
+    pth = pth();
+  }
+
+  pth = normalizeHomedir(pth, ctx.homedir);
+
+  if (!isAbsolutePath(pth)) {
+    return path.join(ctx.cwd, pth);
+  }
+
+  return path.normalize(pth);
+}
+
+
+var initCtx = {
+  homedir: os.homedir ? os.homedir() : require('userhome')(),
+  cwd: process.cwd(),
+};
+
+function normalizeContext(ctx) {
+  ctx = eachProp(ctx || {}, doNothing);
+
+  ctx.cwd = !ctx.cwd ? initCtx.cwd :
+    normalizePath(ctx.cwd, initCtx);
+
+  ctx.homedir = !ctx.homedir ? initCtx.homedir :
+    normalizePath(ctx.homedir, initCtx);
+
+  return ctx;
+}
+
+function normalizeExtensions(exts) {
+  if (isFunction(exts)) {
+    exts = exts();
+  }
+
+  if (isString(exts)) {
+    return [exts];
+  }
+
+  if (isArray(exts)) {
+    return exts.filter(isString);
+  }
+
+  if (isObject(exts)) {
+    return Object.keys(exts).sort();
+  }
+
+  return [''];
+}
+
+function findFile(pth, basedir, findUp) {
+
+  var parsed = parsePath(pth);
+  if (parsed.root) { // pth is absolute path
+    findUp = false;
+    basedir = parsed.root;
+    pth = pth.slice(parsed.root.length);
+  }
+
+  var globOpt = { nocase: true, cwd: basedir };
+
+  if (isWindows()) {
+    pth = pth.replace(/\\/g, '/');
+  }
+
+  if (!findUp) {
+    return eachUnNullElem(glob.sync(pth, globOpt), toFullPath, globOpt.cwd);
```

Comment:
I'm cautiously :+1: on this. I believe that Liftoff is too complex already, but only because some of the config finding logic needs to become a standalone module. If the functionality you are describing were added as a dependent library that Liftoff consumes, this could be very valuable.